integreat_cms/static/src/js/search-query.ts

```diff
@@ -78,36 +88,94 @@ export const setSearchQueryEventListeners = () => {
         );
     });
 
-    const tableSearch = document.getElementById("table-search");
+    tableSearchInput.addEventListener("keydown", (event) => {
+        const suggestions = Array.from(suggestionList?.children || []) as HTMLElement[];
+        const input = event.target as HTMLInputElement;
+
+        if (suggestions.length === 0) {
+            return;
+        }
+
+        if (event.key === "ArrowLeft") {
+            event.preventDefault();
+            if (input.selectionStart !== null && input.selectionEnd !== null) {
+                const newCaretPosition = Math.max(0, input.selectionStart - 1);
+                input.setSelectionRange(newCaretPosition, newCaretPosition);
+            }
+        } else if (event.key === "ArrowRight") {
+            event.preventDefault();
+            if (input.selectionStart !== null && input.selectionEnd !== null) {
+                const newCaretPosition = Math.min(input.value.length, input.selectionEnd + 1);
+                input.setSelectionRange(newCaretPosition, newCaretPosition);
+            }
+        }
 
-    tableSearch.addEventListener("focusout", (_event) => {
-        const searchSuggestion = document.getElementById("table-search-suggestions");
-        searchSuggestion.classList.add("hidden");
+        if (event.key === "ArrowDown") {
+            event.preventDefault();
+            focusedIndex = (focusedIndex + 1) % suggestions.length;
+        } else if (event.key === "ArrowUp") {
+            event.preventDefault();
+            focusedIndex = (focusedIndex - 1 + suggestions.length) % suggestions.length;
+        } else if (event.key === "Enter") {
+            event.preventDefault();
+            const activeSuggestion = suggestions[focusedIndex];
+            if (activeSuggestion) {
+                tableSearchInput.value = activeSuggestion.textContent || "";
+                document.getElementById("search-submit-btn")?.click();
+                suggestionList?.classList.add("hidden");
+                focusedIndex = -1;
+            }
+            return;
```

Comment:
This is very similar to this, only a few lines below:

```js
    suggestionList?.addEventListener("keydown", (event) => {
        if (event.key === "Enter") {
            const activeSuggestion = document.activeElement as HTMLElement;
            if (activeSuggestion && suggestionList.contains(activeSuggestion)) {
                event.preventDefault();
                tableSearchInput.value = activeSuggestion.textContent || "";
                document.getElementById("search-submit-btn")?.click();
                suggestionList?.classList.add("hidden");
            }
        }
    });
```
I think you can simply remove this (old) code - right? At least, when it's commented out, I can't seem to find any differences in behavior

Reply:
@charludo this one `on dropdown list`, without on enter it won't work unfortunately with up and down arrows as well as with tab. The above was` on input field` enter.